api/server/docker.go

```diff
@@ -218,30 +222,60 @@ func (d *driver) handshake(w http.ResponseWriter, r *http.Request) {
 }
 
 func (d *driver) attachToken(ctx context.Context, request *volumeRequest) (context.Context, string) {
-	token, tokenInName := d.GetTokenFromString(request.Name)
-	if !tokenInName {
-		token = request.Opts[api.Token]
+
+	// get token from name
+	tokenFromName, tokenInName := d.GetTokenFromString(request.Name)
+	if tokenInName {
+		return addTokenMetadata(ctx, tokenFromName), tokenFromName
 	}
-	if len(token) == 0 {
-		return ctx, ""
+
+	// get token from opts
+	tokenFromOpts := request.Opts[api.Token]
+	if tokenFromOpts != "" {
+		return addTokenMetadata(ctx, tokenFromOpts), tokenFromOpts
 	}
 
+	// get token secret
+	secret, context, ok := d.GetTokenSecretFromString(request.Name)
+	if ok && d.secretsStore != nil {
+		token, err := d.secretsStore.GetToken(secret, context)
```

Comment:
I think we should return an error when secret store is not initialized, but a token_secret was provided. In this way the user would get the correct error.

In the current format, user would get "not authorized" error instead of "secret store not initialized"